app/src/sharedTest/java/org/oppia/app/topic/TopicFragmentTest.kt

```diff
@@ -0,0 +1,347 @@
+    onView(withId(R.id.topic_tabs_container)).check(matches(matchCurrentTabTitle("PLAY")))
+  }
+
+  @Test
+  @UiThread
+  fun testTopicActivity_clickOnPlayTab_isPlayTabIsSelectedAndContentMatched() {
+    activityTestRule.launchActivity(null)
+    onView(
+      allOf(
+        withText("PLAY"),
+        isDescendantOfA(withId(R.id.topic_tabs_container))
+      )
+    ).perform(click())
+    onView(withText("First Story")).check(matches(isDisplayed()))
+  }
+
+  @Test
+  @UiThread
+  fun testTopicActivity_clickOnTrainTab_isSwitchToTrainTabAndTrainTabIsSelected() {
+    activityTestRule.launchActivity(null)
+    onView(
+      allOf(
+        withText("TRAIN"),
+        isDescendantOfA(withId(R.id.topic_tabs_container))
+      )
+    ).perform(click())
+    onView(withId(R.id.topic_tabs_container)).check(matches(matchCurrentTabTitle("TRAIN")))
+  }
+
+  @Test
+  @UiThread
+  fun testTopicActivity_clickOnTrainTab_isTrainTabIsSelectedAndContentMatched() {
+    activityTestRule.launchActivity(null)
+    onView(
+      allOf(
+        withText("TRAIN"),
+        isDescendantOfA(withId(R.id.topic_tabs_container))
+      )
+    ).perform(click())
+    onView(withText("Master These Skills")).check(matches(isDisplayed()))
+  }
+
+  @Test
+  @UiThread
+  fun testTopicActivity_clickOnReviewTab_isSwitchToReviewTabAndReviewTabIsSelected() {
+    activityTestRule.launchActivity(null)
+    onView(
+      allOf(
+        withText("REVIEW"),
+        isDescendantOfA(withId(R.id.topic_tabs_container))
+      )
+    ).perform(click())
+    onView(withId(R.id.topic_tabs_container)).check(matches(matchCurrentTabTitle("REVIEW")))
+  }
+
+  @Test
+  @UiThread
+  fun testTopicActivity_clickOnReviewTab_isReviewTabIsSelectedAndContentMatched() {
+    activityTestRule.launchActivity(null)
+    onView(
+      allOf(
+        withText("REVIEW"),
+        isDescendantOfA(withId(R.id.topic_tabs_container))
+      )
+    ).perform(click())
+    onView(
+      allOf(
+        atPosition(R.id.review_skill_recycler_view, 0).also { withText("An important skill") },
+        isDescendantOfA(withId(R.id.review_skill_recycler_view))
+      ).also { withId(R.id.skill_name) })
+  }
+
+  @Test
+  @UiThread
+  fun testTopicActivity_showsTopicActivityWithMultipleTabs_clickOnOverviewTab_isTabLayoutDisplayedWithOverviewTabSelected() {
+    activityTestRule.launchActivity(null)
+    onView(
+      allOf(
+        withText("REVIEW"),
+        isDescendantOfA(withId(R.id.topic_tabs_container))
+      )
+    ).perform(click())
+    onView(
+      allOf(
+        withText("OVERVIEW"),
+        isDescendantOfA(withId(R.id.topic_tabs_container))
+      )
+    ).perform(click())
+
+    onView(withId(R.id.topic_tabs_container)).check(matches(matchCurrentTabTitle("OVERVIEW")))
+
+  }
+
+  @Test
+  @UiThread
+  fun testTopicActivity_clickOnOverviewTab_isOverviewTabSelectedAndContentMatched() {
+    activityTestRule.launchActivity(null)
+    onView(
+      allOf(
+        withText("REVIEW"),
```

Comment:
Here and everywhere else, get tab-names from `strings.xml`

Reply:
Resolved